### DISCUSSION TITLE
cheatsheets/vim/

### DISCUSSION THREAD

**Discussion by giscus** [shane0/cheatsheets#2] [General]
# cheatsheets/vim/

https://shane0.github.io/cheatsheets/vim/

**Comment by shane0** [comment]
- macos  https://github.com/pqrs-org/Karabiner-Elements

**Comment by shane0** [comment]
# macos 

I press my option key with my left thumb to turn on vim mode in any application

/Users/shanenull/.config/karabiner

https://github.com/pqrs-org/Karabiner-Elements

[karabiner_shane.json.txt](https://github.com/shane0/cheatsheets/files/12696530/karabiner_shane.json.txt)

https://ke-complex-modifications.pqrs.org/build/extra_descriptions/vim_mode_plus.json.html

<img width="1079" alt="image" src="https://github.com/shane0/cheatsheets/assets/1454458/eb2e1bef-5ae4-4cd6-991a-b10d4f9053fd">

<img width="1085" alt="image" src="https://github.com/shane0/cheatsheets/assets/1454458/9e6c57bf-8159-4a35-b15c-208c2878fc6a">

https://vimium.github.io/

**Comment by shane0** [comment]
# qmk

- i have a few qmk keyboards
- atreus
- 40% and 60% split keebio
- in the past i used a viterbi, planck, keyboardio, ergodox, dactyl

## atreus

- currently use chrysalis & a combo of default ergodox layers + vim layers

[current.pdf](https://github.com/shane0/cheatsheets/files/12696585/current.pdf)

[current.json.txt](https://github.com/shane0/cheatsheets/files/12696589/current.json.txt)

**Comment by shane0** [comment]
# searching
 
**Here's how to search within the same line in Vim and navigate between matches:**

1. **Initiate the search:**
   - Press `/` to initiate a forward search.
   - Press `?` to initiate a backward search.

2. **Type your search pattern:** Enter the text you want to find within the current line.

3. **Search within the line:**
   - Press `t` to jump to the next occurrence of the pattern **before** the cursor.
   - Press `T` to jump to the next occurrence of the pattern **after** the cursor.

4. **Navigate between matches:**
   - Press `n` to repeat the last search in the same direction (forward or backward).
   - Press `N` to repeat the last search in the opposite direction.

**Example:**

- To find the next "foo" after the cursor: `/foo` followed by `n`.
- To find the previous "bar" before the cursor: `?bar` followed by `N`.

**Additional tips:**

- **Case sensitivity:** Use `\c` to ignore case in your search pattern (e.g., `/\cfoo`).
- **Whole word matches:** Wrap your pattern in `\<` and `\>` to match whole words (e.g., `/\<foo\>`).
- **Visual mode:** Select text and press `/` or `?` to search for the selected pattern.
- **Wrap around:** Use the `wrapscan` option to allow searches to wrap around the end of the file (set it with `:set wrapscan`).

**Using `f` and `F` to search within the current line:**

- **`f{char}`:** Jumps to the **next** occurrence of the specified character **on or after** the cursor within the current line.
- **`F{char}`:** Jumps to the **previous** occurrence of the specified character **on or before** the cursor within the current line.

**Key points:**

- Searches only within the current line.
- Doesn't support regular expressions or search patterns, only single characters.
- Useful for quick jumps to specific characters within a line.

**Examples:**

- To jump to the next "a" after the cursor: `fa`
- To jump to the previous "b" before the cursor: `Fb`

**Common use cases:**

- Moving quickly within function names or variable names.
- Jumping to specific parts of a line, like a colon in a dictionary definition.

**Combining with `t` and `T`:**

- After using `f` or `F` to jump to a character, you can use `t` or `T` to jump to the same character on subsequent lines.
  - `ft` jumps to the next occurrence of the same character on the next line.
  - `FT` jumps to the previous occurrence of the same character on the previous line.

**Remember:**

- `/` and `?` are more versatile for general searches within files, supporting patterns and options.
- `f` and `F` are specialized for quick character-based jumps within the current line.

**Yes, you can go both forward and backward after using `f` or `F` to jump to a character. Here's how:**

- **To go forward to the next occurrence of the same character:**
   - Press `;` (semicolon).
- **To go backward to the previous occurrence of the same character:**
   - Press `,` (comma).

**Essentially:**

- `f` or `F` finds the initial character.
- `;` and `,` cycle through subsequent matches of that character within the same line.

**Additional notes:**

- These commands operate within the **current line only**.
- To move forward or backward to the same character on subsequent lines, use `t` or `T` as mentioned earlier.

**Example:**

1. You're on a line with multiple instances of "x".
2. Press `fx` to jump to the next "x".
3. Press `;` to jump to the next "x" after that.
4. Press `,` to return to the previous "x".

**Key point:**

- `;` and `,` always move to the **next or previous occurrence of the same character** you initially searched for with `f` or `F`.
